jaas/src/main/java/com/griddynamics/jagger/jaas/rest/TestEnvironmentRestController.java

```diff
@@ -82,33 +75,33 @@ public TestEnvironmentRestController(TestEnvironmentService testEnvironmentServi
     }
 
     @PutMapping(value = "/{envId}", consumes = APPLICATION_JSON_VALUE)
-    @ApiOperation(value = "Updates Test Environment by envId.", notes = "This operation can be performed only if 'Environment-Session' cookie is "
-            + "present and valid. To obtain this cookie POST to /envs must be performed firstly. This cookie is valid only for Test Environment with "
+    @ApiOperation(value = "Updates Test Environment by envId.", notes = "This operation can be performed only if '"
+            + TestEnvUtils.SESSION_COOKIE + "' cookie is "
+            + "present and valid. To obtain this cookie POST to /envs must be performed firstly. "
+            + "This cookie is valid only for Test Environment with "
             + "envId which was specified in POST request body.")
     @ApiResponses(value = {
             @ApiResponse(code = 202, message = "Update is successful."),
             @ApiResponse(code = 404, message = "Test Environment with provided envId or sessionId is not found."),
-            @ApiResponse(code = 400, message = "Environment-Session cookie is not present.")})
-    public ResponseEntity<?> updateTestEnvironment(@PathVariable String envId, @RequestBody TestEnvironmentEntity testEnv,
-                                                   HttpServletRequest request, HttpServletResponse response) {
+            @ApiResponse(code = 404, message = TestEnvUtils.SESSION_COOKIE + " cookie is not present.")})
+    public ResponseEntity<?> updateTestEnvironment(@CookieValue(TestEnvUtils.SESSION_COOKIE) String sessionId,
+                                                   @PathVariable String envId,
+                                                   @RequestBody TestEnvironmentEntity testEnv) {
         if (!testEnvService.exists(envId))
             throw ResourceNotFoundException.getTestEnvResourceNfe();
 
-        String sessionId = getSessionCookie(request);
-        if (sessionId == null)
-            throw new TestEnvironmentNoSessionException();
         if (!testEnvService.existsWithSessionId(envId, sessionId))
             throw new TestEnvironmentSessionNotFoundException(envId, sessionId);
 
         testEnv.setEnvironmentId(envId);
-        TestEnvironmentEntity updated = testEnvService.update(testEnv);
-        setExpiresHeader(response, updated);
-        setSessionCookie(response, updated, environmentsTtlMinutes * 60);
```

Comment:
Why did you remove it? Cookie Expires won't be updated and it'll expire very soon.